sql/create.go

```diff
@@ -126,7 +126,28 @@ func (p *planner) CreateTable(n *parser.CreateTable) (planNode, error) {
 	// Inherit permissions from the database descriptor.
 	desc.Privileges = dbDesc.GetPrivileges()
 
-	if err := desc.AllocateIDs(); err != nil {
+	err = desc.AllocateIDs()
+	if err == errMissingPrimaryKey {
```

Comment:
I don't follow; you can't do `if len(desc.PrimaryIndex.ColumnNames) == 0`? The check in `Validate` is checking that the IDs were assigned, but you can just look at the names here.